01_fib/baijie/fib_test.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+	expected := strconv.Quote("1\n")
+	actual := strconv.Quote(buf.String())
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main(), expected %v, actual %v", expected, actual)
+	}
+}
+
+func TestFib(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	type test struct {
+		input    int
+		expected string
+	}
+
+	s1 := []string{"1", "1", "2", "3", "5", "8", "13"}
+	s2 := []string{"1", "-1", "2", "-3", "5", "-8", "13"}
+
+	tests := []test{
+		{input: 0, expected: strconv.Quote("")},
+		{input: 1, expected: strconv.Quote("1\n")},
+		{input: -1, expected: strconv.Quote("1\n")},
+		{input: 7, expected: strconv.Quote(strings.Join(s1, "\n") + "\n")},
+		{input: -7, expected: strconv.Quote(strings.Join(s2, "\n") + "\n")},
+	}
+
+	for _, test := range tests {
+		fib(test.input)
+		actual := strconv.Quote(buf.String())
+
+		if test.expected != actual {
+			t.Errorf("Unexpected output in main()\ninput: %d\nexpected: %q\nactual  : %q", test.input, test.expected, actual)
+		}
+		buf.Reset()
```

Comment:
You could avoid having to do a `buf.Reset()` if you pushed lines 24-25 inside the for loop.

Reply:
Do you mean move below lines inside the for loop? 

	var buf bytes.Buffer
	out = &buf

Reply:
Yes. If you moved them inside the loop at the top, each iteration of the loop would start with a fresh buffer that you do not need to `Reset()` at the end of the loop.